src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -2606,8 +2606,8 @@ private bool TryProcessEnvChange(int tokenLength, TdsParserStateObject stateObj,
                                 if (newCodePage != _defaultCodePage)
                                 {
                                     _defaultCodePage = newCodePage;
-                                    _defaultEncoding = System.Text.Encoding.GetEncoding(_defaultCodePage);
                                 }
+                                _defaultEncoding = System.Text.Encoding.GetEncoding(_defaultCodePage);
```

Comment:
This will trigger exception since _defaultCodePage is 0 in this case. This value will only be set when we actually get code page for the server.

Reply:
It actually gets System default encoding (On Windows it's "windows-1252"), hence the test passes :)

Reply:
Sorry, it's my mistake. If you pass 0, it will use the default encoding according to client environment. But it should not be the purpose all across the driver logic since this value should actually be the codepage from server not the client.

Reply:
Yep, we probably have to map all encodings for collations supported by server, that's the bigger picture!

Reply:
Thanks for capturing details Karina. I agree and guess we can't really then default to client encoding and the right solution would only be to map everything in the driver if .NET APIs don't provide us correct CodePage. I'll defer the PR for now, and we can come back to it to fix it properly with full mapping similar to other drivers.